homeassistant/components/light/lifx.py

```diff
@@ -138,6 +143,28 @@ def aiolifx_effects():
                                config,
                                async_add_entities,
                                discovery_info=None):
+    """Set up the LIFX light platform.
+
+    Deprecated.
+    """
+    _LOGGER.warning('Loading LIFX via light platform config is deprecated.')
+    await _async_setup_platform(
+        hass, config, async_add_entities, discovery_info)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up LIFX from a config entry."""
+    try:
+        config = vol.Schema(CONFIG)(hass.data[LIFX_DOMAIN].get('light', {}))
```

Comment:
Validation in component

Reply:
Done. I had to remove the defaults but maybe just as well to keep those in the third party library.